### ISSUE TITLE
Any chance this can be made to work in the Windows terminal?

### ISSUE THREAD

**Issue by SimonArdrey** [derrickpelletier/node-status#9]
It only renders after the task is complete. I ran your example and it does not render the increments.
It's a great looking UI tool, I'd love to be able to use it in a Windows environment if possible.

Thanks :)

**Comment by derrickpelletier** [comment]
Hmmmm, possibly. But I don't have access to a windows machine, myself.
Gonna rename this issue, and maybe someone else will tackle it. Thanks for bringing it up

**Comment by pilsy** [comment]
hey @SimonArdrey are you by any chance running NPM > 2.x ? because on my mac i get the same issue when using NPM 2 @derrickpelletier

**Comment by derrickpelletier** [comment]
Which node version @pilsy?

**Comment by pilsy** [comment]
$ node -v
v0.12.0
$ npm -v
2.5.1

**Comment by SimonArdrey** [comment]
@derrickpelletier I am running 1.4.28

**Comment by derrickpelletier** [comment]
Yeah looks like an issue with node v0.12

**Comment by pilsy** [comment]
Hey @derrickpelletier, Oddly this started to work for me after clearing my npm cache and doing a fresh npm i for my project https://github.com/CleverStack/cleverstack-cli/

**Comment by derrickpelletier** [comment]
Yeah i only go to take a real quick look on the weekend, but it's sporadic. At times it works, and sometimes it stalls. Trying to get a reproducible case to state from anyway.

**Comment by derrickpelletier** [comment]
I've tracked this down to an issue in the `charm` dependency. I think that the charm.position(callback) function seems to hang or not report data for some reason.

Unfortunately charm seems more or less inactive, so it might need to look into custom solution for getting cursor position.

**Comment by pilsy** [comment]
@derrickpelletier is there a github issue for this in https://github.com/substack/node-charm/issues ?

**Comment by derrickpelletier** [comment]
@pilsy thanks for the nudge.

I just tested this on `v0.12.7` and I'm seeing it function perfectly fine. (running the `example.js` in the repo, anyway.

Rolling back to `v0.12.0` i can see that it is still functioning incorrectly on that node version.

I haven't looked into it but i'm assuming there was probably some tty-related issue that got resolved somewhere in between. Can you try a more recent version?

**Comment by pilsy** [comment]
The issue also exists when using io.js (any version) but specifically i am using 
```
node -v
v2.2.1
npm -v
2.11.0
```

**Comment by derrickpelletier** [comment]
I can confirm that the issue exists in iojs v2.2.0
But, it is functioning fine for me in iojs v2.4.0

Can you upgrade and try that out?

**Comment by pilsy** [comment]
Yep that works - ok so what did you want to do with this ticket?

**Comment by derrickpelletier** [comment]
Just going to figure out the exact version breakpoints and put a note in the readme, not going to attempt to workaround the problem

**Comment by cliffordduke** [comment]
This seems to still be an issue, I tried to get it working on my Windows 7 Machine and it's also only rendering after the task is complete.

```
node -v
v4.4.7

npm -v
3.10.5
```

**Comment by derrickpelletier** [comment]
Interesting...

Which version of the module?
And are you running the provided examples or your own code?

**Comment by cliffordduke** [comment]
I'm using the latest version on npm `1.0.0`

I initially tried it with my own code, thinking it was something I was doing wrong I then tried running the `examples/pattern.js` example.  Both had the same issue.

**Comment by derrickpelletier** [comment]
Alright, thanks

Unsure if/when I'll have a windows machine to debug this problem on though. I'll check and see if there are any related known-issues regarding the console on windows though.

**Comment by zchristopoulos** [comment]
Running into the same issue on Windows 10. node v4.5.0, npm v2.15.9

**Comment by twillard** [comment]
Same issue.  Windows 7, running in git-bash mintty

$ node --version
v6.9.1
$ npm --version
4.0.3
$ git --version
git version 2.10.1.windows.1
$ mintty --version
mintty 2.5.0 (x86_64-pc-msys)
(C) 2013/2016 Andy Koppe / Thomas Wolff
License GPLv3+: GNU GPL version 3 or later
There is no warranty, to the extent permitted by law.

**Comment by beth-panx** [comment]
Experiencing the same issue on Windows 10. node v7.7.2, npm v4.1.2

**Comment by derrickpelletier** [comment]
Just a reminder that I do not have a windows machine, nor access to one, and anyone is free to contribute :)

**Comment by azarus** [comment]
There are many problems with this on windows terminals. Probably because you're rerouting the stdout & stdin pipes, and even keyboard doesn't work.

**Comment by frankred** [comment]
On windows machines
```
var status = require('node-status');
....
status.stop()
```
Does never stops sometimes.
